interface cache-scalar {

    /// An error occurred while making the call.
    variant error {
        internal-error,
        request-cancelled,
        invalid-argument(string),
        timeout,
        permission-denied(string),
        limit-exceeded(string),
        failed-precondition(string),
        not-found(string),
    }

    get: func(key: list<u8>) -> result<option<list<u8>>, error>;
    set: func(key: list<u8>, value: list<u8>, ttl-milliseconds: u64) -> result<_, error>;
}
